Fix for Page OnAppearing triggered twice when navigating via ShellItem change with PresentationMode set to Modal

SyedAbdulAzeemSF4852 opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

### Issue Details

- When navigating to a new ShellItem using a route like //SomeRoute/SomePage, and SomePage specifies Shell.PresentationMode="Modal", the page’s OnAppearing event is called twice.

### Root Cause

- During Shell navigation between different ShellItems (e.g., //tab1 to //tab2/modal), modal pages receive duplicate lifecycle events because the old ShellSection calls SendDisappearing() on the modal page during its cleanup, but the modal page now belongs to the new ShellSection, which also manages its lifecycle events.


### Description of Change

- Added an IsVisibleSection check in PresentedPageDisappearingmethod  to ensure lifecycle events are only sent by the active ShellSection. This prevents duplicate events on modal pages during ShellItem transitions by blocking non-visible (previous) ShellSections from managing pages they no longer own.


### Issues Fixed
Fixes #31584 

### Validated the behaviour in the following platforms

- [x] Windows
- [x] Android
- [x] iOS
- [x] Mac

### Output
| Platform | Before | After |
|----------|----------|----------|
| iOS | <video src="https://github.com/user-attachments/assets/eef8275d-7c08-402e-9370-3382801d1715"> | <video src="https://github.com/user-attachments/assets/2bacf7f7-0e66-4000-aa9d-1f65601e3877"> |
| Android | <video src="https://github.com/user-attachments/assets/706dbef4-9c5c-4c50-8bb4-da7addc89527"> | <video src="https://github.com/user-attachments/assets/77845043-b12e-4efe-9c3e-0bc7a27da318"> |

SyedAbdulAzeemSF4852 committed:
Fix for Page OnAppearing triggered twice when navigating via ShellItem change with PresentationMode set to Modal

PureWeen commented:
/rebase

PureWeen commented:
/azp run

PureWeen reviewed (changes requested):
Can you rework this to just be a unit test vs a UItest? 

I think we can validate this purely by being a UITest?

PureWeen committed:
- generate tests with copilot

SyedAbdulAzeemSF4852 committed:
Removed the UI test since unit tests have been added, and also removed duplicate tests in the unit tests

SyedAbdulAzeemSF4852 commented:
> [Conversation_Summary_PR_32582.md](https://github.com/user-attachments/files/23615394/Conversation_Summary_PR_32582.md)
> 
> @SyedAbdulAzeemSF4852 @sheiksyedm
> 
> I had copilot add unit tests based on your UITest
> 
> I asked it this
> 
> > "The PR here #32582 is using a UITest can you instead make this a unit test? Please make sure to validate that the test fails without the changes and then it's fixed with the changes. Feel free to add any additional tests."
> 
> and then it generated these tests
> 
> Please review added tests and delete UITests if they look good

@PureWeen , I’ve gone through the tests Copilot added. Two of the tests seemed quite similar and were essentially testing the same thing, so I’ve removed one of them. I’ve also removed the original UI test since the remaining unit tests provide the necessary coverage.

PureWeen commented:
/rebase
